modules/core/src/main/java/org/apache/fluo/core/impl/FluoConfigurationImpl.java

```diff
@@ -101,6 +101,44 @@ public static int getTxCommitMemory(FluoConfiguration conf) {
     }
     return m;
   }
+  
+  // Note the size of the cache is in megabytes
+  public static final String TX_INFO_CACHE_SIZE = FLUO_IMPL_PREFIX + ".tx.failed.cache.size.mb";
+  /* TODO: What is a reasonable default size? Currently CacheBuilder does not specify a size in TxInfoCache.java */
+  public static final long TX_INFO_CACHE_SIZE_DEFAULT = 1;
+  
+  /** 
+   * Gets the value of the property {@ #TX_INFO_CACHE_SIZE} if set,
+   * else gets the default value {@ #TX_INFO_CACHE_SIZE_DEFAULT}
+   * 
+   * @param conf
+   * @return The size of the cache in megabytes
+   */
+  public static long getTxInfoCacheSize(FluoConfiguration conf) {
+    long size = conf.getLong(TX_INFO_CACHE_SIZE, TX_INFO_CACHE_SIZE_DEFAULT);
+    if (size <= 0) {
+      throw new IllegalArgumentException("Cache size must be positive for " + TX_INFO_CACHE_SIZE);
+    }
+    return size;
+  }
+  
+  public static final String TX_INFO_CACHE_TIMEOUT = FLUO_IMPL_PREFIX + ".tx.failed.cache.expireTime";
+  public static final int TX_INFO_CACHE_TIMEOUT_DEFAULT = 24 * 60; 
+  
+  /**
+   * Gets the cache expire time that entrys will be evicted after if not accessed
+   * 
+   * @param conf
+   * @return The shelf life of entries before they will be evicted if not accessed
+   * TODO: What timeunit would make most sense? Currently TxCacheInfo uses TimeUnit.MINUTES
```

Comment:
Thats a very good question.  I looked at some time props in FluoConfiguration and the timeunits are mostly well documented because of your previous change.  Need to document the timeunit for zookeeper timeout, this could be a new help wanted issue, would you like to open that?  

I did find one prop with a pattern I really liked in FluoConfiguration

```java
public FluoConfiguration setTransactionRollbackTime(long time, TimeUnit tu) {
    return setPositiveLong(TRANSACTION_ROLLBACK_TIME_PROP, tu.toMillis(time));
  }
```

I like two things about this pattern :
  * In the java API it uses timeunit, so  that makes the code very clear and explicit.
  * Internally it converts everything to milliseconds and the prop name ends with `ms`.  

Something that transaction rollback time didn't do that would have been nice would be to accept a timeunit on the getter.  For cache timeout, this could be something like `long getTxInfoCacheTimeout(FluoConfiguration conf, TimeUnit tu)` and it will return the timeout in the specified unit.

This makes me think it would be nice to have a SizeUnit, I am not sure if this is a good idea or not.  May be worthwhile opening an issue to discuss.  Could make config API nicer.